src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -974,12 +976,9 @@ void primitive_inst::do_runtime_skip_reorder() {
                 update_memory_dependencies(u->get_user_insts());
 
                 u->set_can_be_optimized(true);
-                // Opt out reorder which has _needs_completion_event = true causes syncronization failed in dGPU.
-                if (_needs_completion_event == false && u->_needs_completion_event == true) {
-                    _needs_completion_event = true;
-                }
                 GPU_DEBUG_TRACE_DETAIL << "[do runtime skip reorder] set user " << u->id() << " as can_be_optimized" << std::endl;
             } else {
+                u->set_can_be_optimized(false);
```

Comment:
random spot: please add an additional condition of is_runtime_skippable  at line948, because we should not reset can_be_opt as false if the true flag is set by other condition, not by the runtime skip.

Reply:
@yeonbok, You mean add the condition like this?
`    if (can_be_optimized() || get_node().is_runtime_skippable())`

Reply:
not || but &&.

Reply:
Done. please check it again. thanks.